app/controllers/admin/schedules_controller.rb

```diff
@@ -4,72 +4,50 @@ class SchedulesController < Admin::BaseController
     load_and_authorize_resource :program, through: :conference, singleton: true
+    load_and_authorize_resource :schedule, through: :program
+    load_resource :event_schedules, through: :schedule
+    load_resource :selected_schedule, through: :program, singleton: true
     load_resource :venue, through: :conference, singleton: true
 
-    skip_before_filter :verify_authenticity_token, only: [:update]
-    layout 'schedule'
+    skip_before_action :verify_authenticity_token, only: [:update]
 
-    def show
-      authorize! :update, @program.events.new
-      if @conference.nil?
-        redirect_to admin_conference_index_path
-        return
-      end
-      @dates = @conference.start_date..@conference.end_date
-      if @venue && @venue.rooms.any?
-        @rooms = @venue.rooms
-      else
-        @rooms = [ Room.new(name: 'No Rooms!', size: 0) ]
-      end
+    def index
     end
 
-    def update
-      authorize! :update, @program.events.new
-      event = Event.where(guid: params[:event]).first
-      error_message = nil
-      if event.nil?
-        error_message = "Could not find event GUID: #{params[:event]}"
+    def create
+      if @schedule.save
+        redirect_to action: 'show', id: @schedule.id
+      else
+        flash[:error] = 'Could not create schedule'
+        redirect_to action: 'index'
       end
+    end
 
-      if params[:date] == 'none'
-        event.start_time = nil
-        event.room = nil
-        event.save!
-        render json: { 'status' => 'ok' }
-        return
-      end
-      room = Room.where(guid: room_params).first
-      if room.nil?
-        error_message = "Could not find room GUID: #{params[:room]}"
-      end
+    def show
+      @event_schedules = @schedule.event_schedules
+      @unscheduled_events = @program.events.confirmed - @schedule.events
+      @dates = @conference.start_date..@conference.end_date
+      @rooms = (@venue && @venue.rooms.any?) ? @venue.rooms : [Room.new(name: 'No Rooms!', size: 0)]
+    end
 
-      unless error_message.nil?
-        render json: { 'status' => 'error', 'message' => error_message }, status: 500
-        return
+    def update
+      if params[:selected_schedule] == 'true'
+        @program.selected_schedule_id = params[:id].to_i
+      elsif params[:selected_schedule] == 'false' && (@selected_schedule.id == params[:id].to_i)
+        @program.selected_schedule_id = nil
       end
-
-      event.room = room
-      time = "#{params[:date]} #{params[:time]}"
-
-      Rails.logger.debug("Loading #{time}")
-      # FIXME: Same here as in events_controller.rb. Event timezone should be applied
-      # only on output
-      # zone = ActiveSupport::TimeZone::new(@conference.timezone)
-      # start_time = DateTime.strptime(time + zone.formatted_offset, "%Y-%m-%d %k:%M %Z")
-      start_time = DateTime.strptime(time, '%Y-%m-%d %k:%M')
-      event.start_time = start_time
-      event.save!
+      @program.save!
```

Comment:
And what I predicted happens in our heroku app ;)
Because the new migration didn't run, there is no @program.selected_schedule_id field and you get a 500.

Please don't use the bang and render an error if sth goes wrong!